libnavui-maps/src/main/java/com/mapbox/navigation/ui/maps/route/line/model/MapboxRouteLineOptions.kt

```diff
@@ -215,10 +242,35 @@ class MapboxRouteLineOptions private constructor(
          * Enabling this feature will result in route legs that aren't currently being navigated
          * to be color differently than the active leg. See [RouteLineColorResources] for the
          * color option.
+         *
+         * @return the builder
          */
         fun styleInactiveRouteLegsIndependently(enable: Boolean): Builder =
             apply { this.styleInactiveRouteLegsIndependently = enable }
 
+        /**
+         * Influences the appearance of the gradient used for the traffic line. By default
+         * there is an abrupt transition between different colors representing traffic congestion.
+         * If this value is set to true there is a smoother transition between the colors. This
+         * value is false by default.
+         *
+         * @return the builder
+         */
+        fun displaySoftGradientForTraffic(enable: Boolean): Builder =
+            apply { this.displaySoftGradientForTraffic = enable }
+
+        /**
+         * When [displayRestrictedRoadSections] is set to true this value will influence the
+         * length of the gradient transition between traffic congestion colors. Larger values
+         * will result in the color transition occurring over a longer length of the line.
+         * Values between 5 and 75 are recommended. The default value is 30.0
+         *
+         * @param transitionDistance a value above zero used for the gradient transition distance
```

Comment:
Do we need to check at creation if the value is above zero and return an `IllegalArgumentException` if not?

Reply:
Yes.